src/components/ProofOfDeliveryModal.vue

```diff
@@ -0,0 +1,406 @@
+import { closeOutline, saveOutline } from "ionicons/icons";
+import { useStore } from "vuex";
+import { showToast } from "@/utils";
+import { hasError } from "@/adapter";
+import { translate } from "@hotwax/dxp-components";
+import { OrderService } from "@/services/OrderService";
+
+
+export default defineComponent({
+  name: "ProofOfDeliveryModal",
+  components: {
+    IonButtons,
+    IonButton,
+    IonContent,
+    IonHeader,
+    IonToolbar,
+    IonIcon,
+    IonTitle,
+    IonItem,
+    IonLabel,
+    IonInput,
+    IonFooter,
+    IonSelect,
+    IonSelectOption,
+    IonCheckbox,
+    IonSpinner
+  },
+  props: {
+    order: { type: Object, required: true },
+    deliverShipmentFn: { type: Function, required: true }
+  },
+  setup(props) {
+    const store = useStore();
+    const isSubmitting = ref(false);
+    const sameAsBilling = ref(false);
+    
+    const form = ref({
+      name: "",
+      idNumber: "",
+      relationToCustomer: "Self",
+      email: ""
+    });
+
+    const billingDetails = ref({});
+
+    // Computed property for form validation
+    const isFormValid = computed(() => {
+      return form.value.name.trim() !== "" && 
+             form.value.idNumber.trim() !== "" && 
+             form.value.relationToCustomer !== "";
+    });
+
+    // Fetch billing details on modal open
+    const getBillingDetails = async () => {
+      try {
+        const resp = await OrderService.getBillingDetails({ orderId: props.order.orderId });
+        billingDetails.value = resp?.data?.billToAddress || {};
+
+        // Prefill form fields from billing details by default
+        form.value.name = billingDetails.value?.toName || billingDetails.value?.contactName || "";
+        form.value.email = billingDetails.value?.email || billingDetails.value?.toEmail || "";
+        form.value.relationToCustomer = "Self"; // Set default relation
+      } catch (err) {
+        console.error("Error fetching billing details:", err);
+      }
+    };
+
+    // Handle Same as Billing checkbox
+    const handleSameAsBilling = () => {
+      if (sameAsBilling.value) {
+        // When checked, restore billing details and prefilled ID
+        form.value.name = billingDetails.value?.toName || billingDetails.value?.contactName || "";
+        form.value.email = billingDetails.value?.email || billingDetails.value?.toEmail || "";
+        form.value.relationToCustomer = "Self";
+      } else {
+        // When unchecked, clear only name and email
+        form.value.name = "";
+        form.value.email = "";
+      }
+    };
+
+    // Fetch order attributes to get prefield id for customerId and prefill idNumber
+    const getPrefilledValue = async () => {
+      try {
+        if (!props.order?.orderId) return;
+        const resp = await OrderService.fetchOrderAttributes(props.order.orderId);
+        const data = resp?.data ?? resp;
+        const list = Array.isArray(data) ? data : (data?.docs || data?.attributes || []);
```

Comment:
Is there any case when resp.data will not be available or resp.data.docs won't be available?